src/main/java/com/google/sps/servlets/GetElectionServlet.java

```diff
@@ -0,0 +1,169 @@
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+@WebServlet("/getElection")
+
+/**
+ * This servlet is used to update the gVote database with information on the ongoing elections that
+ * an eligible voter can participate in on a given day.
+ */
+public class GetElectionServlet extends HttpServlet {
+
+  private static final String baseURL = "https://www.googleapis.com/civicinfo/v2/elections?key=";
+  private static final Logger logger = Logger.getLogger(ElectionServlet.class.getName());
+
+  // This method is used to access the api key stored in gcloud secret manager.
+  public String accessSecretVersion(String projectId, String secretId, String versionId)
+      throws IOException {
+    try (SecretManagerServiceClient client = SecretManagerServiceClient.create()) {
+      SecretVersionName secretVersionName = SecretVersionName.of(projectId, secretId, versionId);
+
+      // Access the secret version.
+      AccessSecretVersionResponse response = client.accessSecretVersion(secretVersionName);
+
+      // Get the secret payload
+      return response.getPayload().getData().toStringUtf8();
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Deleting the queries from yesterday in the case that they are outdated
+    Query query = new Query("Election");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      Key electionEntityKey = KeyFactory.createKey("Election", entity.getKey().getId());
+      datastore.delete(electionEntityKey);
+    }
+
+    StringBuilder strBuf = new StringBuilder();
+    HttpURLConnection conn = null;
+    BufferedReader reader = null;
+
+    try {
+      // Running on a local server requires storing API key in environment variable.
+      // In this case, uncomment the following line, and comment out the call to
+      // accessSecretVersion. Do the reverse when deploying.
+      // String electionApiKey = System.getenv("GOOGLE_API_KEY");
```

Comment:
This is a pretty low level implementation detail. Maybe have a function getApiKey() that abstracts away which one is happening. Then move the comment above into that functin.

Also something to look into would be if you can control this via a flag or something. Leave a TODO about figuring out via a flag?